srcCpp/perftest_publisher.cxx

```diff
@@ -67,20 +68,30 @@ int main(int argc, char *argv[])
 }
 
 #if defined(RTI_VXWORKS)
-int publisher_main()
-{
-    const char *argv[] = {"perftest_cpp", "-pub"};
-    int argc = sizeof(argv)/sizeof(const char*);
+int perftest_cpp_main(char* args) {
+    std::vector<char*> arguments;
 
-    return main(argc, (char **) argv);
-}
+    // split args by " " and add each one to dynamic array
+    char* next = strtok(args, " ");
+    while (next != NULL) {
+		arguments.push_back(next);
+		next = strtok(NULL, " ");
+    } 
 
-int subscriber_main()
-{
-    const char *argv[] = {"perftest_cpp", "-sub"};
-    int argc = sizeof(argv)/sizeof(const char*);
+    // Copy dynamic array to the original
+    // NOTE: +1 because Run() expects also the executable name argv[0].
+    //          we skip the first index
+    int argc = arguments.size()+1;
+    char *argv[argc];
+
+    argv[0] = "perftest_cpp"; // Dummy name for the executable
```

Comment:
What about if you add this to the vector at the beginning? wouldn't that simplify the code? (Not sure, just asking)

Reply:
Good idea :+1: